base/linalg/triangular.jl

```diff
@@ -1879,19 +1888,20 @@ function sqrtm{T}(A::UpperTriangular{T})
             end
         end
     end
+    sqrtm(A,Val{realmatrix})
+end
+function sqrtm{T,realmatrix}(A::UpperTriangular{T},::Type{Val{realmatrix}})
     if realmatrix
         TT = typeof(sqrt(zero(T)))
     else
         TT = typeof(sqrt(complex(-one(T))))
```

Comment:
just use `sqrt(complex(zero(T)))` here, since at some point `one` may return a dimensionless value if `T` is dimensionful.